.github/workflows/build.yml

```diff
@@ -2,12 +2,22 @@ name: Python Fire
 
 on: [push]
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version: ["3.5", "3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
+        fail-fast: false
```

Comment:
```suggestion
        fail-fast: true
```

Reply:
I recommend `false`, this way you get to see what exactly failed -- is it all versions lower than 3.10? Is it Windows?

With `true`, if just the 3.5/Windows job failed, and all the others were cancelled, you don't get this useful info.

Reply:
true is by default, just make it explicit, I normally prefer to run all but it could yield quite extensive testing

Reply:
```suggestion
```